db/silkworm/common/log.hpp

```diff
@@ -17,48 +17,49 @@
 #ifndef SILKWORM_COMMON_LOG_H_
 #define SILKWORM_COMMON_LOG_H_
 
-#include <absl/time/clock.h>
-
 #include <silkworm/common/tee.hpp>
+#include <mutex>
 
 namespace silkworm {
 
-enum LogLevels { LogTrace, LogDebug, LogInfo, LogWarn, LogError, LogCritical, LogNone };
+// stream labeled logging output - e.g.
+//	  SILKWORM_LOG(LogInfo) << "All your " << num_bases << " base are belong to us\n";
+//
+#define SILKWORM_LOG(level_) if ((level_) < log_verbosity_()) {} else log_(level_)
 
-// Log to one or two output streams - typically the console and optional log file.
-class Logger {
-  public:
-    explicit Logger(std::ostream& o1 = std::cerr, std::ostream& o2 = Logger::null_stream()) : stream_(o1, o2) {}
+// change the logging verbosity level - default level is LogInfo
+//
+#define SILKWORM_LOG_VERBOSITY(level_) log_verbosity_(level_)
 
-    LogLevels verbosity{LogInfo};
-
-    void set_local_timezone(bool local) noexcept { timezone_ = local ? absl::LocalTimeZone() : absl::UTCTimeZone(); }
+// available verbosity levels
+enum LogLevels { LogTrace, LogDebug, LogInfo, LogWarn, LogError, LogCritical, LogNone };
 
-    std::ostream& log(LogLevels level) {
-        return stream_ << kLogTags[level] << "[" << absl::FormatTime("%m-%d|%H:%M:%E3S", absl::Now(), timezone_)
-                       << "] ";
+// change the logging output streams - default is (cerr, null_stream())
+//
+#define SILKWORM_LOG_STREAMS(stream1_, stream2_) log_set_streams_((stream1_), (stream2_));
+
+// silence
+std::ostream& null_stream();
+
+//
+// Below are for access via macros ONLY :(
+//
+LogLevels log_verbosity_();
```

Comment:
I'd wrap stuff that's not part of the API into `detail` namespace (like they do in Boost).

Reply:
Good idea.  Looks like this will our first use of that namespace.

Reply:
It's used in `db/silkworm/db/util.hpp` already :)